src/Optimizer/Transformer/PreloadHeroImage.php

```diff
@@ -563,6 +568,18 @@ private function generateImg(HeroImage $heroImage, Document $document)
         $imgElement->setAttribute(Attribute::CLASS_, self::SSR_IMAGE_CLASS);
         $imgElement->setAttribute(Attribute::DECODING, 'async');
 
+
+        // If the image was detected as hero image candidate (and thus lacks an explicit data-hero), mark it as a hero
+        // and add loading=lazy to guard against making the page performance even worse by eagerly loading an image
+        // outside the viewport.
+        if (! $this->isMarkedAsHeroImage($element)) {
+            $imgElement->setAttribute(Attribute::LOADING, 'lazy');
+        }
```

Comment:
Maybe there should be an `else` case here to see if there is a `noscript > img` maybe this should copy the `loading` attribute from that image if it is set.

Reply:
I'm creating a separate issue for this, as I want the tests to be fixed again.
See https://github.com/ampproject/amp-toolbox-php/issues/147